pkg/cmd/server/server.go

```diff
@@ -278,12 +295,15 @@ func newServer(f client.Factory, config serverConfig, logger *logrus.Logger) (*s
 		discoveryClient:       veleroClient.Discovery(),
 		dynamicClient:         dynamicClient,
 		sharedInformerFactory: informers.NewSharedInformerFactoryWithOptions(veleroClient, 0, informers.WithNamespace(f.Namespace())),
-		ctx:                   ctx,
-		cancelFunc:            cancelFunc,
-		logger:                logger,
-		logLevel:              logger.Level,
-		pluginRegistry:        pluginRegistry,
-		config:                config,
+		// If no namespace is specified, all namespaces are watched.
+		// This is desirable for VolumeSnapshots, as we want to query for all VolumeSnapshots across all namespaces using this informer
+		snapshotterSharedInformerFactory: snapshotvebeta1informers.NewSharedInformerFactoryWithOptions(csiSnapClient, 0),
```

Comment:
I think we may want to consider wrapping this instantiation, and all subsequent uses of the informer factory, in a feature-flag check too

Reply:
I actually had a function that did something similar before I pushed up any of this, but it was based on gating access to the clients. I'll see about tweaking it to add it for the informer factory.